src/catalog/src/durable/impls/persist.rs

```diff
@@ -1146,7 +1156,7 @@ impl PersistHandle {
     {
         let (snapshot, current_upper) = self.current_snapshot().await;
         let next_upper = current_upper.step_forward();
-        let trace = Trace::from_snapshot(snapshot);
+        let trace = Trace::from_snapshot(snapshot.clone());
```

Comment:
We could probably avoid this clone with a lot of typing, but this is for the catalog-debug CLI which is hardly used and the latency difference would probably be unnoticeable to a human.